payment/src/main/java/org/killbill/billing/payment/core/PaymentMethodProcessor.java

```diff
@@ -414,7 +415,7 @@ public void deletedPaymentMethod(final Account account, final UUID paymentMethod
                     try {
                         // Note: account.getPaymentMethodId() may be null
                         if (paymentMethodId.equals(account.getPaymentMethodId())) {
-                            if (!deleteDefaultPaymentMethodWithAutoPayOff) {
+                            if (!deleteDefaultPaymentMethodWithAutoPayOff && !forceDefaultPaymentMethodDeletion) {
                                 throw new PaymentApiException(ErrorCode.PAYMENT_DEL_DEFAULT_PAYMENT_METHOD, account.getId());
                             } else {
                                 final boolean isAccountAutoPayOff = isAccountAutoPayOff(account.getId(), context);
```

Comment:
I think we are missing a `if` condition here (below the `else`):

```
} else {
    if (deleteDefaultPaymentMethodWithAutoPayOff && !isAccountAutoPayOff(account.getId(), context)) {
        log.info("Setting AUTO_PAY_OFF on accountId='{}' because of default payment method deletion", account.getId());
         setAccountAutoPayOff(account.getId(), context);
    }
    accountInternalApi.removePaymentMethod(account.getId(), context);
}
```